alf/utils/dist_utils.py

```diff
@@ -604,33 +595,57 @@ def _get_independent_builder(obj: td.Independent):
     return new_builder, params
 
 
-def _get_transform_params(transforms):
+def _get_transform_builders_params(transforms):
     """Return a nested structure where each node is a non-composed transform,
     after expanding any composed transform in ``transforms``.
     """
-    if isinstance(transforms, td.Transform):
-        if isinstance(transforms, td.ComposeTransform):
-            return _get_transform_params(transforms.parts)
-        elif hasattr(transforms, 'params') and transforms.params is not None:
+
+    def _get_transform_builder(transform):
+        if hasattr(transform, "get_builder"):
+            return transform.get_builder()
+        return transform.__class__
+
+    def _get_transform_params(transform):
+        if hasattr(transforms, 'params') and transforms.params is not None:
             # We assume that if a td.Transform has attribute 'params', then they are the
             # parameters we'll extract and store.
+            assert isinstance(
+                transforms.params,
+                dict), ("Transform params must be provided as a dict! "
+                        f"Got {transforms.params}")
             return transforms.params
-        else:  # the transform doesn't have any parameter
-            return ()
+        return ()  # the transform doesn't have any parameter
+
+    if isinstance(transforms, td.Transform):
+        if isinstance(transforms, td.ComposeTransform):
+            builders, params = _get_transform_builders_params(transforms.parts)
+            compose_transform_builder = lambda params: td.ComposeTransform(
+                [b(p) for b, p in zip(builders, params)])
+            return compose_transform_builder, params
+        else:
+            builder = _get_transform_builder(transforms)
+            params = _get_transform_params(transforms)
+            new_builder = lambda params: (builder(**params)
+                                          if params != () else builder())
+            return new_builder, params
```

Comment:
can be simplified as the following if line 617 is changed to `return {}` and `b(p)` at 574  to `b(**p)`:
```python
return builder, params
```

Reply:
> can be simplified as the following if line 617 is changed to `return {}` and `b(p)` at 574 to `b(**p)`:
> 
> ```python
> return builder, params
> ```

Updated